src/FHIR/SMART/SMARTLaunchToken.php

```diff
@@ -98,13 +99,14 @@ public function serialize()
         if (!empty($intent)) {
             $context['i'] = $intent;
         }
+        $cryptoGen = new CryptoGen();
+        $jsonEncoded = json_encode($context);
+        $launchParams = $cryptoGen->encryptStandard($jsonEncoded);
 
         // no security is really needed here... just need to be able to wrap
         // the current context into some kind of opaque id that the app will pass to the server and we can then
         // return to system
-        // TODO: adunsulag do we want a nonce here? don't think it will matter as user has to pass through oauth2 grant
-        // in order to get back the launch code.
-        $serialized = base64_encode(json_encode($context));
+        $serialized = base64_encode($launchParams);
```

Comment:
note that encryptStandard() will return base64_encoded and decryptStandard() will assume the input is base64_encoded